core/modules/drr.cc

```diff
@@ -274,21 +303,21 @@ void DRR::AddNewFlow(bess::Packet* pkt, FlowId id, int* err) {
   Flow* f = new Flow(id);
 
   // TODO(joshua) do proper error checking
-  f->queue = AddQueue(static_cast<int>(kFlowQueueSize), err);
+  f->queue = generator_();
```

Comment:
ah, I guess the main point of the factory is that it binds the parameters to the generator so that DRR doesn't have to have a place to store each queue type's parameters. I could just store these parameters and use this code if you think that is better? but I could also help simplify things down as panda suggested and have the factory take the protobuf message instead?